modules/vstudio/vs2010_vcxproj.lua

```diff
@@ -1746,6 +1746,8 @@
 				m.element("LanguageStandard", condition, 'stdcpp17')
 			elseif (cfg.cppdialect == "C++20") then
 				m.element("LanguageStandard", condition, iif(_ACTION == "vs2017", 'stdcpplatest', 'stdcpp20'))
+			elseif (cfg.cppdialect == "C++23") then
+				m.element("LanguageStandard", condition, 'stdcpp23')
```

Comment:
We should probably figure out a better, long term way of handling this, but for now, please change `'stdcpp23'` here to the `iif` statement above. I believe these statements should likely be `<= 'vs2017'` for C++20 and `<= 'vs2019'` for C++23.

In addition, this will require an additional test to be written in the vstudio module.

Reply:
I'll make the change but I'm curious what is the point of it? For example you might request C++23 on vs2017 and because you get c++latest instead, your code does not compile perhaps in a non-obvious way?
Wouldn't it be better to explicitly use C++23 and let vs2017 give an error if it does not support it? (Or have premake spit an error, but it's harder to know exactly what vs supports).

Reply:
> I'm curious what is the point of it?

Consistency and scope. As @nickclark2016 said, we need to figure out a better, long term way of handling this. Your argument also applies to C++20 and vs2017, so it goes beyond the scope of this PR. Getting C++23 support in is a good first step, and we can just log an issue to deal with this remapping stuff.

Reply:
I just noticed that vs2022 doesn't actually support stdcpp23 yet, so we should probably change it to use stdcpplatest (ie. replace the <= 2019 by < 2022).

If that sounds OK, I can a new PR tonight.

Reply:
You could probably just make it emit `stdcpplatest`, either works. Does MSC support C++23, or will that also need to be changed?

Reply:
Oh that's a good catch. It does not.

Reply:
PR #2281 changes both vs2022 and MSC